kong/router/compat.lua

```diff
@@ -330,38 +332,44 @@ end
 -- split routes into multiple routes, one for each prefix length and one for all
 -- regular expressions
 local function split_route_by_path_into(route_and_service, routes_and_services_split)
-  if is_empty_field(route_and_service.route.paths) or #route_and_service.route.paths == 1 then
-    table.insert(routes_and_services_split, route_and_service)
+  local original_route = route_and_service.route
+
+  if is_empty_field(original_route.paths) or #original_route.paths == 1 then
+    tb_insert(routes_and_services_split, route_and_service)
     return
   end
 
   -- make sure that route_and_service contains only the two expected entries, route and service
-  assert(tablex.size(route_and_service) == 2)
+  assert(tb_nkeys(route_and_service) == 2)
 
   local grouped_paths = group_by(
-    route_and_service.route.paths,
+    original_route.paths,
     function(path)
       return is_regex_magic(path) or #path
     end
   )
   for index, paths in pairs(grouped_paths) do
     local cloned_route = {
-      route = utils.shallow_copy(route_and_service.route),
+      route = utils.shallow_copy(original_route),
       service = route_and_service.service,
     }
-    cloned_route.route.original_route = route_and_service.route
+
+    cloned_route.route.original_route = original_route
     cloned_route.route.paths = paths
-    cloned_route.route.id = uuid_generator(route_and_service.route.id .. "#" .. tostring(index))
-    table.insert(routes_and_services_split, cloned_route)
+    cloned_route.route.id = uuid_generator(original_route.id .. "#" .. tostring(index))
+
+    tb_insert(routes_and_services_split, cloned_route)
   end
 end
 
 
 local function split_routes_and_services_by_path(routes_and_services)
   local routes_and_services_split = tb_new(#routes_and_services, 0)
-  for i = 1, #routes_and_services do
-    split_route_by_path_into(routes_and_services[i], routes_and_services_split)
+
+  for _, v in ipairs(routes_and_services) do
+    split_route_by_path_into(v, routes_and_services_split)
   end
+
```

Comment:
why this change?

Reply:
I think `ipairs` may be better optimized by LuaJIT.